src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/randomized/ConditionsForTwoPhaseOptimization.java

```diff
@@ -0,0 +1,18 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.randomized;
+
+public class ConditionsForTwoPhaseOptimization {
```

Comment:
I don't think you need an explicit class for that. Instead, simply make the two conditions two separate arguments of the constructor of `TwoPhaseQueryOptimizer`.